File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/LineValue.java
package com.flat.simplex.parser.logic.line;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;
import com.flat.simplex.parser.logic.line.call.LineCall;

public class LineValue extends Line {

    public enum ReturnType {
        Boolean, Double, String, Object, Void
    }

    public LineValue(Block parent, Token token) {
        super(parent, token);
    }

    public ReturnType getReturnType() {
        return ReturnType.Object;
    }

    public LineCall getFirstCall() {
        return null;
    }

    public LineCall getLastCall() {
        return null;
    }

    @Override
    public boolean isOp() {
        return false;
    }

    @Override
    public LineValue getValue() {
        return this;
    }

    public boolean isContainer() {
        return false;
    }
}

File: SimplexApp/simplex/parser/logic/line/LineValue.js
const Line = require("simplex/parser/logic/line/Line.js");

class LineValue extends Line {
    constructor(parent, token) {
        super(parent, token);
    }

    getReturnType() {
        return LineValue.Object;
    }

    getFirstCall() {
        return null;
    }

    getLastCall() {
        return null;
    }

    isOp() {
        return false;
    }

    getValue() {
        return this;
    }

    isContainer() {
        return false;
    }
}

LineValue.Boolean = 'Boolean';
LineValue.Double = 'Double';
LineValue.String = 'String';
LineValue.Object = 'Object';
LineValue.Void = 'Void';

module.exports = LineValue;